src/test_typing_extensions.py

```diff
@@ -4681,23 +4709,39 @@ class AllTheThings(TypedDict):
             },
         )
 
-    def test_extra_keys_non_readonly(self):
+    @skipIf(TYPING_3_14_0, "Old syntax only supported on <3.14")
+    def test_extra_keys_non_readonly_compat(self):
         class Base(TypedDict, closed=True):
             __extra_items__: str
 
-        class Child(Base):
-            a: NotRequired[int]
+        with self.assertWarns(DeprecationWarning):
+            class Child(Base):
+                a: NotRequired[int]
 
         self.assertEqual(Child.__required_keys__, frozenset({}))
         self.assertEqual(Child.__optional_keys__, frozenset({'a'}))
         self.assertEqual(Child.__readonly_keys__, frozenset({}))
         self.assertEqual(Child.__mutable_keys__, frozenset({'a'}))
 
+    @skipIf(TYPING_3_14_0, "Only supported on <3.14")
     def test_extra_keys_readonly(self):
         class Base(TypedDict, closed=True):
             __extra_items__: ReadOnly[str]
 
-        class Child(Base):
+        with self.assertWarns(DeprecationWarning):
+            class Child(Base):
+                a: NotRequired[str]
+
+        self.assertEqual(Child.__required_keys__, frozenset({}))
+        self.assertEqual(Child.__optional_keys__, frozenset({'a'}))
+        self.assertEqual(Child.__readonly_keys__, frozenset({}))
+        self.assertEqual(Child.__mutable_keys__, frozenset({'a'}))
+
+    def test_extra_keys_readonly_explicit_closed(self):
+        class Base(TypedDict, closed=True):
+            __extra_items__: ReadOnly[str]
+
+        class Child(Base, closed=True):
             a: NotRequired[str]
```

Comment:
`__required_keys__` and `__optional_keys__` will contain `__extra_itmes__` when using 3.14

```suggestion
    @skipIf(TYPING_3_14_0, "Only supported on <3.14")
    def test_extra_keys_readonly_explicit_closed(self):
        class Base(TypedDict, closed=True):
            __extra_items__: ReadOnly[str]

        class Child(Base, closed=True):
            a: NotRequired[str]
```